js/pop-up.js

```diff
@@ -0,0 +1,139 @@
+import projects from './project-list.js';
+
+const detailsTargetElement = document.querySelector('#details-target-element');
+const worksTargetElement = document.querySelector('#works-target-element');
+
+const generateDetailsWork = (
+  id,
+  title,
+  tags,
+  image,
+  imageAlt,
+  fullDescription,
+  frameworks,
+  seeLiveLink,
+  seeSourceLink,
+) =>
+  `<div class="details">
+  <article id="${id}" class="details-target-element details-target-element details-works-article details-popup left">
+<div class="works-article details-work-article d-block left">
+  <div class="details-header">
+    <h3 class="works-headline details-works-headline">
+      ${title}
+      <svg id="close-details-element" class="close-details absolute" width="30" height="30" viewBox="0 0 25 29" fill="none"
+        xmlns="http://www.w3.org/2000/svg">
+        <path fill-rule="evenodd" clip-rule="evenodd"
+          d="M5.29289 5.29289C5.68342 4.90237 6.31658 4.90237 6.70711 5.29289L12 10.5858L17.2929 5.29289C17.6834 4.90237 18.3166 4.90237 18.7071 5.29289C19.0976 5.68342 19.0976 6.31658 18.7071 6.70711L13.4142 12L18.7071 17.2929C19.0976 17.6834 19.0976 18.3166 18.7071 18.7071C18.3166 19.0976 17.6834 19.0976 17.2929 18.7071L12 13.4142L6.70711 18.7071C6.31658 19.0976 5.68342 19.0976 5.29289 18.7071C4.90237 18.3166 4.90237 17.6834 5.29289 17.2929L10.5858 12L5.29289 6.70711C4.90237 6.31658 4.90237 5.68342 5.29289 5.29289Z"
+          fill="black" />
+      </svg>
+    </h3>
+  </div>
+  <ul class="details-about-icon d-flex">
+  ${tags
+    .map((tag) => `<li class="details-content">${tag}</li>`)
+    .join('<li class="dot details-dot"></li>')}
+  </ul>
+
+  <div class="img-profile details-img-profile">
+    <img class="details-works-img" src="${image}" alt="${imageAlt}" />
+  </div>
+
+  <p class="works-writeup details-works-writeup d-block-lg">
+  ${fullDescription}
+  </p>
+
+  <ul class="abt-link details-abt-link fixed-lg d-flex">
+${frameworks
+  .map((framework, index) =>
+    index < 3
+      ? `<li class="works-link details-works-link">${framework}</li>`
+      : `<li class="works-link details-works-link d-none d-block-lg">${framework}</li>`,
+  )
+  .join('')}
```

Comment:
and please can you explain why we have to handle this situation this way?

Reply:
I made use of [Ternary Operator](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwj5pLryuomDAxXMV0EAHb9XCmUQFnoECAUQAQ&url=https%3A%2F%2Fdeveloper.mozilla.org%2Fen-US%2Fdocs%2FWeb%2FJavaScript%2FReference%2FOperators%2FConditional_operator&usg=AOvVaw2jzewi8EwBigYAdAIdjnX6&opi=89978449), it reduce code and aids understanding

Reply:
i'm not talking about the ternary operator in particular ,i just don't understand why you had to check if the values of frame work is less than 3 before you assign the right thing when you already have an array that handles the whole thing for you and its dynamic enough i feel you can just assign it straight

Reply:
@Yinkusadel Thanks for the review, And you are right, I could have assign it straight, but I decided to make use of ternary operator, because I want the framework to display only 3 when in mobile screen and display 5 when in large screen